Itemlink tutorial improvement (#611)
ArchipelagoMW/Archipelago: worlds/generic/docs/advanced_settings_en.md

* `exclude_locations` lets you define any locations that you don't want to do and during generation will force a "junk"
  item which isn't necessary for progression to go in these locations.

> * `item_links` allows players to link their items into a group with the same item link name and game. The items get combined and when an item link item is found all players receive it. Item links can also have local and non local items. Also if someone in the item link has something missing from the pool that someone else in the item link has and it is supposed to be linked it will not be linked.

Reviewer: ```suggestion
* `item_links` allows players to link their items into a group with the same item link name and game. The items declared in `item_pool` get combined and when an item is found for the group, all players in the group receive it. Item links can also have local and non local items, forcing the items to either be placed within the worlds of the group or in worlds outside the group. If players have a varying amount of a specific item in the link, the lowest amount from the players will be the amount put into the group.
```

Reviewer 2: You did nothing with this feedback. Either address it directly or take the suggestion. Resolving a conversation without a word is discouraged.

Reviewer: He actually did sort of write what i suggested but differently then I ended up PRing it in with different wording and you've critiqued some of my writing now 😄

description: An example using various advanced options
name: Example Player
game: 

>   A Link to the Past 10

Reviewer: ```suggestion
  A Link to the Past: 10
```

  Timespinner: 10
requires: 

>   version: 0.2.0

Reviewer: ```suggestion
  version: 0.3.2
```
since this is using item links this also needs to be updated for consistency.

        bigkey_shuffle: any_world
        map_shuffle: any_world
        compass_shuffle: any_world
Timespinner:
  item_links: # Share part of your item pool with other players.
    - name: TSAll
      item_pool: 
        - Everything
      local_items:
        - Twin Pyramid Key
        - Timespinner Wheel
      replacement_item: null 
    
```

#### This is a fully functional yaml file that will do all the following things:

* `description` gives us a general overview so if we pull up this file later we can understand the intent.
* `name` is `Example Player` and this will be used in the server console when sending and receiving items.

> * `game` has an equal chance of being either `A Link to the Past` or `Timespinner` with a 10/20 chance for each.

Reviewer 2: The sentence says they have an equal chance, then that they each have 10/20 chance. Instead of restating chances, please describe in one clause that they are equal 10/20 chances.

        bigkey_shuffle: any_world
        map_shuffle: any_world
        compass_shuffle: any_world

> Timespinner:

Reviewer: Lines 195-281 should be removed, since they don't add any value.

* `description` gives us a general overview so if we pull up this file later we can understand the intent.
* `name` is `Example Player` and this will be used in the server console when sending and receiving items.
* `game` has an equal chance of being either `A Link to the Past` or `Timespinner` with a 10/20 chance for each. The reason for this is becuase each game has a weight of 10 and the toal of all weights is 20.

> * `requires` is set to require release version 0.3.2 or higher.

Reviewer 2: ```suggestion
* `requires` is set to required release version 0.3.2 or higher.
```

* `start_location_hints` gives us a starting hint for the `Spike Cave` location available at the beginning of the
  multiworld that can be used for no cost.
* `exclude_locations` forces a not important item to be placed on the `Cave 45` location.
* `item_links` 
  * For `A Link to the Past` all players in the `rods` item link group will share their fire and ice rods and the player
items will be replaced with single rupees.

>   * For `Timespinner` all players in the `TSAll` item link group will share their entire item pool and the`Twin Pyramid 

Reviewer 2: Missing space between "the" and "`Twin Pyramid Key`"

Key` and `Timespinner Wheel` will be forced among the worlds of those in the group. The `null` replacement item instead

> of forcing a filler item will allow the generator to fill the item.

Reviewer 2: > The `null` replacement item instead of forcing a filler item will allow the generator to fill the item.

The last sentence here doesn't make sense to me. Please elaborate or re-write.

Author: I worded it a bit different to fix it

  * For `A Link to the Past` all players in the `rods` item link group will share their fire and ice rods and the player
items will be replaced with single rupees.
  * For `Timespinner` all players in the `TSAll` item link group will share their entire item pool and the`Twin Pyramid 
  * For `A Link to the Past` all players in the `rods` item link group will share their fire and ice rods and the player
items will be replaced with single rupees.
  * For `Timespinner` all players in the `TSAll` item link group will share their entire item pool and the `Twin Pyramid 

> Key` and `Timespinner Wheel` will be forced among the worlds of those in the group. The `null` replacement item instead

Reviewer 2: ```suggestion
Key` and `Timespinner Wheel` will be forced among the worlds of those in the group. The `null` replacement item will, instead
```

> of forcing a specific chossen item this will allow the generator to randomaly pick a filler/junk/noraml class item in place of putting in another one of the linked item.

Reviewer 2: ```suggestion
of forcing a specific chosen item, allow the generator to randomly pick a filler item in place of putting in another one of the linked item.
```

* `start_location_hints` gives us a starting hint for the `Spike Cave` location available at the beginning of the
  multiworld that can be used for no cost.
* `exclude_locations` forces a not important item to be placed on the `Cave 45` location.

> * `item_links` causes all players with the same `item_links` settings to share a `Fire Rod` and `Ice Rod`. Extra

Reviewer: The Link to the Past and Timespinner item_links are two separate sections and should be documented as such.

Reviewer: this was not fixed correctly.

Author: I think I fixed it now. Let me know if there is any other way to improvements that you suggest.
